packages/core/elasticsearch/core-elasticsearch-server/src/client/scoped_cluster_client.ts

```diff
@@ -22,6 +22,17 @@ export interface IScopedClusterClient {
    * on behalf of the internal Kibana user.
    */
   readonly asInternalUser: ElasticsearchClient;
+
+  /**
+   * A {@link ElasticsearchClient | client} to be used to query the elasticsearch cluster
+   * with the internal Kibana user as primary auth and the current user as secondary auth
+   * (using the `es-secondary-authorization` header).
+   *
+   * Note that only a subset of Elasticsearch APIs support secondary authentication, and that only those endpoints
+   * should be called with this client.
```

Comment:
> with the internal Kibana user as primary auth and the current user as secondary auth (using the `es-secondary-authorization` header).

Reply:
If by current user you mean current active user, then they have no control in which client is used (similar to how they do not get to choose if we're using the internal one of the primary-scoped one).

API consumer will be able to switch between those three clients the same way they did between the two original ones (`clusterClient.asInternalUser`, `clusterClient.asCurrentUser` and `clusterClient.asSecondaryAuthUser`)